src/pubsub/ua_pubsub_writergroup.c

```diff
@@ -903,15 +903,8 @@ generateNetworkMessage(UA_PubSubConnection *connection, UA_WriterGroup *wg,
 
     networkMessage->version = 1;
     networkMessage->networkMessageType = UA_NETWORKMESSAGE_DATASET;
-    if(connection->config->publisherIdType == UA_PUBSUB_PUBLISHERID_NUMERIC) {
-        networkMessage->publisherIdType = UA_PUBLISHERDATATYPE_UINT16;
-        networkMessage->publisherId.publisherIdUInt32 =
-            connection->config->publisherId.numeric;
-    } else if(connection->config->publisherIdType == UA_PUBSUB_PUBLISHERID_STRING) {
-        networkMessage->publisherIdType = UA_PUBLISHERDATATYPE_STRING;
-        networkMessage->publisherId.publisherIdString =
-            connection->config->publisherId.string;
-    }
+    networkMessage->publisherIdType = connection->config->publisherIdType;
+    networkMessage->publisherId = connection->config->publisherId;
```

Comment:
At connection creation (UA_Server_addPubSubConnection()) a deep copy of the publisherId has been performed.
Then the Id should be available until connection is destroyed.
So it should be ok to reuse it here.

I have actually had some issues with the string PublisherIds -> but after deep copy at UA_Server_addPubSubConnection() everything worked, and I also checked pointers, and allocated configuration on stack which went out of scope and PubSub operation still worked, because Ids are copied once and then are available until teardown